lib/tpm2_openssl.c

```diff
@@ -708,27 +708,27 @@ static bool load_public_ECC_from_key(EC_KEY *k, TPM2B_PUBLIC *pub) {
     TPM2B_ECC_PARAMETER *X = &pub->publicArea.unique.ecc.x;
     TPM2B_ECC_PARAMETER *Y = &pub->publicArea.unique.ecc.y;
 
-    unsigned x_size = BN_num_bytes(x);
+    unsigned x_size = (EC_GROUP_get_degree(group) + 7) / 8;
     if (x_size > sizeof(X->buffer)) {
         LOG_ERR("X coordinate is too big. Got %u expected less than or equal to"
                 " %zu", x_size, sizeof(X->buffer));
         goto out;
     }
 
-    unsigned y_size = BN_num_bytes(y);
+    unsigned y_size = (EC_GROUP_get_degree(group) + 7) / 8;
     if (y_size > sizeof(Y->buffer)) {
         LOG_ERR("X coordinate is too big. Got %u expected less than or equal to"
                 " %zu", y_size, sizeof(Y->buffer));
         goto out;
     }
 
-    X->size = BN_bn2bin(x, X->buffer);
+    X->size = BN_bn2binpad(x, X->buffer, x_size);
     if (X->size != x_size) {
```

Comment:
This function isn't available on openssl < 1.1. you'll have to add a compat routine until we drop 1.0.2 support.

Reply:
Thank you. I added a static compat function with `#if OPENSSL_VERSION_NUMBER < 0x10100000L`.